src/plugins/data/public/ui/ui_service.ts

```diff
@@ -43,6 +45,10 @@ export class UiService implements Plugin<IUiSetup, IUiStart> {
         if (enhancements.query && enhancements.query.language) {
           this.queryEnhancements.set(enhancements.query.language, enhancements.query);
         }
+        if (enhancements.queryEditorExtension) {
```

Comment:
Should this maybe be?
```suggestion
        if (enhancements.queryEditorExtension?.id) {
```

Reply:
id is enforced on the type level, is type checking not enough and we prefer to also add runtime checking in osd?